yt_dlp/extractor/servus.py

```
@@ -15,130 +12,97 @@ class ServusIE(InfoExtractor):
-            })
-        access_token = token['access_token']
-        token_type = token.get('token_type', 'Bearer')
-
         video = self._download_json(
-            'https://sparkle-api.liiift.io/api/v1/stv/channels/international/assets/%s' % video_id,
-            video_id, 'Downloading video JSON', headers={
-                'Authorization': '%s %s' % (token_type, access_token),
-            })
-
-        formats = []
-        thumbnail = None
-        for resource in video['resources']:
-            if not isinstance(resource, dict):
-                continue
-            format_url = url_or_none(resource.get('url'))
-            if not format_url:
-                continue
-            extension = resource.get('extension')
-            type_ = resource.get('type')
-            if extension == 'jpg' or type_ == 'reference_keyframe':
-                thumbnail = format_url
-                continue
-            ext = determine_ext(format_url)
-            if type_ == 'dash' or ext == 'mpd':
-                formats.extend(self._extract_mpd_formats(
-                    format_url, video_id, mpd_id='dash', fatal=False))
-            elif type_ == 'hls' or ext == 'm3u8':
-                formats.extend(self._extract_m3u8_formats(
-                    format_url, video_id, 'mp4', entry_protocol='m3u8_native',
-                    m3u8_id='hls', fatal=False))
-            elif extension == 'mp4' or ext == 'mp4':
-                formats.append({
-                    'url': format_url,
-                    'format_id': type_,
-                    'width': int_or_none(resource.get('width')),
-                    'height': int_or_none(resource.get('height')),
-                })
+            'https://api-player.redbull.com/stv/servus-tv?timeZone=Europe/Berlin&videoId=%s' % video_id,
+            video_id, 'Downloading video JSON')
+        if 'videoUrl' not in video:
+            self._report_errors(video)
+            return None
```

> [pukkandan](https://github.com/pukkandan) [on Jul 10, 2022](https://github.com/yt-dlp/yt-dlp/pull/4240#discussion_r917382328)
return None is invalid. You should raise an exception instead